src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -737,10 +757,25 @@ public boolean hasntStartedYet() {
         return changeSets;
     }
 
-    /** Replacement for {@link SCM#getEffectiveBrowser} to work around JENKINS-35098. TODO 2.7.3+ delete */
-    private static RepositoryBrowser<?> getEffectiveBrowser(SCM scm) {
-        RepositoryBrowser<?> b = scm.getBrowser();
-        return b != null ? b : scm.guessBrowser();
+    @Override
+    @CheckForNull public Set<String> getCulpritIds() {
+        if (shouldCalculateCulprits()) {
+            HashSet<String> tempCulpritIds = new HashSet<>();
+            for (User u : getCulprits()) {
+                tempCulpritIds.add(u.getId());
+            }
+            if (isBuilding()) {
+                return ImmutableSortedSet.copyOf(tempCulpritIds);
+            } else {
+                culprits = ImmutableSortedSet.copyOf(tempCulpritIds);
+            }
+        }
+        return culprits;
+    }
+
+    @Override
+    public boolean shouldCalculateCulprits() {
+        return isBuilding() || culprits == null;
```

Comment:
Is it affected by the issue in Jenkins core API, when `isBuilding()` effectively check only the last run and does not care about others? Maybe be a problem for parallel runs